src/debugger/src/cli_debugger.cpp

```diff
@@ -0,0 +1,83 @@
+#include "debugger/cli_debugger.h"
+#include <iostream>
+
+using namespace debugger;
+using namespace std;
+
+CliDebugger::CliDebugger(std::shared_ptr<DebuggerCore> debugger_core)
+    : debugger_core(debugger_core) {}
+
+void CliDebugger::tick() {
+	cout << "Enter here: ";
+	getline(std::cin, str);
+	bool status = parse_command(str);
+	if (!status) {
+		cout << "Not a valid Command, press h for help" << endl;
+		tick();
+	}
```

Comment:
The name `parse_command` here is a little bit misleading, since it is not just parsing the command, but also executing the actions. `run_command` might be a better name.

Also, this could be readable as an iterative `do..while` loop than as a recursive call. A finer detail is also that since this is a potentially unbounded recursion, if the user kept inputting wrong commands, you'd eventually overflow the call stack :)

Reply:
Yeah, you're right. I guess it slipped my mind. Sorry!